### PR TITLE
Gdb 3888 extract services and utilities

### PR DESCRIPTION
**Pull request by svilenvelikov**
Extract new services and utilities:
* Renamed workbench-core to error-utils and moved in utils package.
* Moved angularCancelOnNavigate module in core package.
* Moved repositories service to core to prevent module dependencies.
* Extracted and moved jwtAuth service and unauthorized interceptor in the core package.
* Moved query editor related directives in the core package.
* Fix paginations module name.
* Extract constants.
* Moved the packaging info in the guide.
* Extracted dependencies service endpoint in a rest client.
* Extract rest service for license info.
* Hide the angular's localStorageService behind LocalStorageAdapter implementation and extracted constants for all used keys.
* Fixing sonar issues.

Jira: https://ontotext.atlassian.net/browse/GDB-3888

### CODE REVIEW

File: test-cypress/integration/repository/repositories.spec.js [hunk 1]
```diff
@@ -22,7 +22,7 @@ describe('Repositories', () => {
         getCreateRepositoryButton().should('be.visible').and('not.be.disabled');
     }
 
-    it('should allow creation of repositories with default settings', () => {
+    it.only('should allow creation of repositories with default settings', () => {
```
Comment (mihailradkov) [inline test-cypress/integration/repository/repositories.spec.js]: Revert?

File: test-cypress/integration/explore/similarity.spec.js [hunk 2]
```diff
@@ -433,14 +433,7 @@ describe('Similarity screen validation', () => {
     }
 
     function setNewQuery(newQuery) {
```
Comment (mihailradkov) [inline test-cypress/integration/explore/similarity.spec.js]: There is no point in having the function now.

File: src/js/angular/utils/local-storage-adapter.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
+angular
+    .module('graphdb.framework.utils.localstorageadapter', [])
+    .factory('LocalStorageAdapter', LocalStorageAdapter)
+    .value('LSKeys', {
+        'TABS_STATE': 'tabs-state',
+        'TABS_STATE_CURRENT_ID': 'tabs-state-currentid',
+        'TABS_STATE_MAXID': 'tabs-state-maxid',
+        'VIEW_MODE': 'viewMode',
+        'CLASS_HIERARCHY_HIDE_PREFIXES': 'classHierarchy-hidePrefixes',
```
Comment (mihailradkov) [inline src/js/angular/utils/local-storage-adapter.js]: Ideally, these should be in their corresponding modules I guess.

File: test-cypress/integration/repository/repositories.spec.js (hunk 1 above)
Comment (svilenvelikov) [reply test-cypress/integration/repository/repositories.spec.js]: pff thx I'll

File: src/js/angular/utils/local-storage-adapter.js (hunk 3 above)
Comment (svilenvelikov) [reply src/js/angular/utils/local-storage-adapter.js]: Yeah, although many of these keys are used in multiple modules, so I either should have them duplicated in every module or I should introduce a way to provide/plug them which I guess might lead to some timing/synchronization issues.

**Pull request merged**